cfme/tests/infrastructure/test_esx_direct_host.py

```diff
@@ -13,14 +13,16 @@
 
 
 def pytest_generate_tests(metafunc):
-    argnames, argvalues, idlist = testgen.provider_by_type(metafunc, ['virtualcenter'])
+    argnames, argvalues, idlist = testgen.providers_by_type(metafunc, ['virtualcenter'])
     argnames = argnames + ["_host_provider"]
 
     new_idlist = []
     new_argvalues = []
 
     for i, argvalue_tuple in enumerate(argvalues):
         args = dict(zip(argnames, argvalue_tuple))
+        # TODO
+        # All this should be replaced with a proper ProviderFilter passed to testgen.providers()
```

Comment:
This isn't a TODO I planned to do in this PR, I just wanted to point out that someone should fix this eventually.